core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java

```diff
@@ -380,7 +387,9 @@ else if (fileChanges.truncate)
           }
         }
       }
-
+      for (OWALRecord externalLog : externalLogs) {
+        writeAheadLog.log(externalLog);
```

Comment:
Write ahead log can be null, this call can produce NPE, you should move it one line above.